flatfs.go

```diff
@@ -540,6 +565,18 @@ func (fs *Datastore) putMany(data map[datastore.Key][]byte) error {
 		dirsToSync = append(dirsToSync, dir)
 
 		tmp, err := fs.tempFile()
+
+		// Fallback retry for temporary error.
+		if err != nil && isTooManyFDError(err) {
```

Comment:
I think we actually need to finish putting the _open_ files here. Otherwise, if there are multiple callers, they'll just all wait on each other to finish the put, then all bail.

Anyways, we should probably have some kind of "I have over 100 files open, time to write them to disk" logic.

Reply:
We can do this in this patch, or a new patch. Up to you.